### PR TITLE
Add PathUtilities.CombinePaths with same semantics as Path.Combine

### PR DESCRIPTION
**Pull request by tmat**
Also adds tests.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler PTAL

File: src/Compilers/Core/Portable/FileSystem/PathUtilities.cs [hunk 1]
```diff
@@ -436,6 +436,33 @@ public static string CombinePathsUnchecked(string root, string? relativePath)
...
+        /// </summary>
+        /// <param name="root">First path: absolute, relative, or null.</param>
+        /// <param name="path">Second path: absolute, relative, or null.</param>
+        /// <returns>
+        /// The combined paths. If <paramref name="path"/> contains an absolute path, returns <paramref name="root"/>.
+        /// </returns>
+        /// <remarks>
+        /// Relative and absolute paths treated the same as <see cref="Path.Combine(string, string)"/>.
+        /// </remarks>
+        public static string? CombinePaths(string? root, string? path)
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/FileSystem/PathUtilities.cs:451]:
```suggestion
       [NotNullIfNotNull("path")]
        public static string? CombinePaths(string? root, string? path)
```

Believe this accurately captures the semantics of the method.

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/Core/Portable/FileSystem/PathUtilities.cs [hunk 2]
```diff
@@ -436,6 +436,34 @@ public static string CombinePathsUnchecked(string root, string? relativePath)
...
         }
 
+        /// <summary>
+        /// Combines paths with the same semantics as <see cref="Path.Combine(string, string)"/>
+        /// but does not throw on null paths or paths with invalid characters.
+        /// </summary>
+        /// <param name="root">First path: absolute, relative, or null.</param>
+        /// <param name="path">Second path: absolute, relative, or null.</param>
+        /// <returns>
+        /// The combined paths. If <paramref name="path"/> contains an absolute path, returns <paramref name="root"/>.
```
Comment (chsienki) [inline src/Compilers/Core/Portable/FileSystem/PathUtilities.cs:446]: Doesn't it return `path` when it's an absolute path?

**Review by chsienki — APPROVED** [review]

File: src/Compilers/Core/Portable/FileSystem/PathUtilities.cs (hunk 2 above)
Comment (tmat) [reply src/Compilers/Core/Portable/FileSystem/PathUtilities.cs:446]: Yes.

**Pull request merged**